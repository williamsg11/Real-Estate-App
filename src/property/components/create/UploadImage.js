/**
 * @flow
 */
import React, {Component, PropTypes} from 'react';
import {
  Image,
  ListView,
  StyleSheet,
  TouchableHighlight,
  View,
} from 'react-native';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import colors from '../../../common/colors';
import ImagePicker from 'react-native-image-crop-picker';
import map from 'lodash/map';

export default class UploadImage extends Component {
  static propTypes = {
    images: PropTypes.array.isRequired,
    updateImage: PropTypes.func.isRequired,
  };

  pickImage = () => {
    const uploadedImages = this.props.images;
    const maxImages = 5;

    ImagePicker.openPicker({
      multiple: true,
    })
      .then(collection => {
        return map(collection, image => image.path);
      })
      .then(images => {
        if (uploadedImages.length >= maxImages) return;
        let allowedImages = [];
        let i = 1;
        images.forEach(image => {
          if (i + uploadedImages.length <= maxImages) {
            allowedImages.push(image);
          }
          i++;
        });
        return allowedImages;
      })
      .then(pendingImages => this.props.updateImage(pendingImages))
      .catch(e => {});
  };

  removeImage = removedImage => {
    this.props.updateImage(removedImage);
  };

  renderRow = (image, key) => {
    return (
      <View style={styles.row}>
        <TouchableHighlight
          onPress={() => this.removeImage(image)}
          underlayColor="transparent"
          style={{
            position: 'absolute',
            zIndex: 1000,
            top: -10,
            left: -10,
          }}>
          <FontAwesome
            name="close"
            style={{
              backgroundColor: 'transparent',
            }}
            color="red"
            size={25}
          />

        </TouchableHighlight>
        <View style={{flex: 1}}>
          <Image key={key} source={{uri: image}} style={styles.image} />
        </View>
      </View>
    );
  };

  render() {
    const {images, header, footer} = this.props;

    let ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 != r2});
    let dataSource = ds.cloneWithRows(images);

    return (
      <View style={styles.container}>

        {header}

        <TouchableHighlight
          style={styles.cameraIcon}
          onPress={() => this.pickImage()}
          underlayColor="transparent">
          <FontAwesome name="camera" size={100} color={colors.white} />
        </TouchableHighlight>

        <View style={styles.menuContainer}>

          <ListView
            dataSource={dataSource}
            contentContainerStyle={styles.contentContainer}
            style={styles.listStyle}
            enableEmptySections={true}
            renderRow={this.renderRow}
            automaticallyAdjustContentInsets={false}
            showsVerticalScrollIndicator={false}
            contentInset={{bottom: 50}}
          />

        </View>

        {images.length > 0 && footer}

      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.smokeGreyLight,
  },
  contentContainer: {
    justifyContent: 'center',
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  listStyle: {
    padding: 5,
  },
  menuContainer: {
    flex: 3,
    padding: 10,
    backgroundColor: 'white',
  },
  row: {
    justifyContent: 'center',
    padding: 5,
    margin: 5,
    width: 140,
    height: 140,
    backgroundColor: '#F6F6F6',
    alignItems: 'center',
    borderWidth: 0.5,
    borderRadius: 2,
    borderColor: '#CCC',
  },
  cameraIcon: {
    flex: 1,
    alignItems: 'center',
    paddingBottom: 20,
  },
  image: {
    width: 125,
    height: 125,
  },
});
