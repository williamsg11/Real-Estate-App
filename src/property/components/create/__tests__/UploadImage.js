import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import UploadImage from './../UploadImage';

test('renders UploadImage', () => {
  const tree = renderer
    .create(<UploadImage images={['a.png']} updateImage={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
