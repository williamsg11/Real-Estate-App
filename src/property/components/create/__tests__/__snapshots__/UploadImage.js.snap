// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UploadImage 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e3e7ed",
      "flex": 1,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "paddingBottom": 20,
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 100,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 3,
        "padding": 10,
      }
    }
  >
    <RCTScrollView
      alwaysBounceHorizontal={undefined}
      alwaysBounceVertical={true}
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
        }
      }
      contentInset={
        Object {
          "bottom": 50,
        }
      }
      dataSource={
        ListViewDataSource {
          "items": 1,
        }
      }
      enableEmptySections={true}
      onContentSizeChange={null}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={undefined}
      onResponderTerminationRequest={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onScrollShouldSetResponder={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      renderRow={[Function]}
      renderScrollComponent={[Function]}
      sendMomentumEvents={false}
      showsVerticalScrollIndicator={false}
      style={
        Array [
          Object {
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "overflow": "scroll",
          },
          Object {
            "padding": 5,
          },
        ]
      }
    >
      <RCTScrollContentView
        collapsable={false}
        removeClippedSubviews={undefined}
        style={
          Array [
            undefined,
            Object {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F6F6F6",
              "borderColor": "#CCC",
              "borderRadius": 2,
              "borderWidth": 0.5,
              "height": 140,
              "justifyContent": "center",
              "margin": 5,
              "padding": 5,
              "width": 140,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "left": -10,
                  "position": "absolute",
                  "top": -10,
                  "zIndex": 1000,
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "red",
                    "fontSize": 25,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "a.png",
                }
              }
              style={
                Object {
                  "height": 125,
                  "width": 125,
                }
              }
            />
          </View>
        </View>
      </RCTScrollContentView>
    </RCTScrollView>
  </View>
</View>
`;
